bugbug/repository.py

```diff
@@ -434,12 +434,14 @@ def download_commits(repo_dir, date_from):
 
             files_touched[path] += 1
 
-        directories = set(
-            directory
-            for path in commit.files
-            for directory in os.path.dirname(path).split("/")[:2]
-            if directory != ""
-        )
+        directories = set()
+        for path in commit.files:
+            path_dirs = (
+                os.path.dirname(path).split("/")[:2] if os.path.dirname(path) else []
+            )
+            if path_dirs:
+                directories.update([path_dirs[0], "/".join(path_dirs)])
```

Comment:
I think we should definitely extract this part in a separate function that takes a path and return a list of directories or an empty list